server/account/account.go

```diff
@@ -13,8 +17,65 @@ const (
 
 type AccountID [HashSize]byte
 
-func New(pk [pki.PubKeySize]byte) AccountID {
+// NewID generates a unique account id with the provided public key bytes.
+func NewID(pk [pki.PubKeySize]byte) AccountID {
 	// hash the pubkey hash(hash(pubkey))
 	h := HashFunc(pk[:])
 	return HashFunc(h[:])
 }
+
+// Account represents a dex account.
+type Account struct {
+	ID      AccountID
+	PrivKey *secp256k1.PrivateKey
+}
+
+// NewAccount creates a new dex account. The associated private key of the
+// account is generated randomly.
+func NewAccount() (*Account, error) {
+	privKey, err := secp256k1.GeneratePrivateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	pubKey := privKey.PubKey()
+
+	var pkb [pki.PubKeySize]byte
+	copy(pkb[:], pubKey.SerializeCompressed()[:pki.PubKeySize])
+	return &Account{
+		ID:      NewID(pkb),
+		PrivKey: privKey,
+	}, nil
+}
+
+// NewAccountFromBytes create a new dex account from the provided private
+// key bytes.
+func NewAccountFromBytes(privK []byte) (*Account, error) {
```

Comment:
This relates to what buck said about the private key vs credentials, but my personal preference here is to be less ambiguous about the function name.  That is, `NewAccountFromPrivKey` explains a bit better what it's doing.  In some constructors `...FromBytes` is perfectly clear (e.g. `secp256k1.PrivKeyFromBytes`), but I would argue this is not one of those cases.